plugin/builtin/sortordertransformer/SortOrderTransformer.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+	"sigs.k8s.io/yaml"
+)
+
+// Sort the resources using a customizable ordering based of Kind.
+// Defaults to the ordering of the GVK struct, which puts cluster-wide basic
+// resources with no dependencies (like Namespace, StorageClass, etc.) first,
+// and resources with a high number of dependencies
+// (like ValidatingWebhookConfiguration) last.
+type plugin struct {
+	SortOptions *types.SortOptions `json:"sortOptions,omitempty" yaml:"sortOptions,omitempty"`
+	// TODO: We can't actually be 100% sure that this isn't the top-level
+	// kustomization, as that information is not propagated through
+	// PluginHelpers. Why not? Because propagating this information would
+	// require extensive code changes and changes to the public API, so we opt
+	// to start small. For now, the transformer will only be called explicitly
+	// if the appropriate fields are set in the kustomization.yaml file, which
+	// will also set the IsTopLevel field.
+	//
+	// See: https://github.com/kubernetes-sigs/kustomize/pull/4019#issuecomment-1159771417
+	IsTopLevel bool
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+// Nothing needed for configuration.
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+
+	// If config was called then this is not the root
```

Comment:
I'd remove this comment. If config was called, then we need to configure the plugin with the incoming data from c, regardless of where we came from.